src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -237,13 +256,116 @@ public static class AkkaClusterHostingExtensions
+        /// optionally, the <see cref="ClusterSingletonProxy"/> created by this method.</param>
+        /// <param name="actorProps">The underlying actor type. SHOULD NOT BE CREATED USING <see cref="ClusterSingletonManager.Props"/></param>
+        /// <param name="options">Optional. The set of options for configuring both the <see cref="ClusterSingletonManager"/> and
+        /// optionally, the <see cref="ClusterSingletonProxy"/>.</param>
+        /// <param name="createProxyToo">When set to <c>true></c>, creates a <see cref="ClusterSingletonProxy"/> that automatically points to the <see cref="ClusterSingletonManager"/> created by this method.</param>
+        /// <typeparam name="TKey">The key type to use for the <see cref="ActorRegistry"/> when <see cref="createProxyToo"/> is set to <c>true</c>.</typeparam>
+        /// <returns>The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.</returns>
+        public static AkkaConfigurationBuilder WithSingleton<TKey>(this AkkaConfigurationBuilder builder,
+            string singletonName, Props actorProps, ClusterSingletonOptions options = null, bool createProxyToo = true)
+        {
+            return builder.WithActors((system, registry) =>
+            {
+                options ??= new ClusterSingletonOptions();
+                var clusterSingletonManagerSettings =
+                    ClusterSingletonManagerSettings.Create(system).WithSingletonName(singletonName);
+
+                var singletonProxySettings =
+                    ClusterSingletonProxySettings.Create(system).WithSingletonName(singletonName);
+
+                if (!string.IsNullOrEmpty(options.Role))
+                {
+                    clusterSingletonManagerSettings = clusterSingletonManagerSettings.WithRole(options.Role);
+                    singletonProxySettings = singletonProxySettings.WithRole(options.Role);
+                }
+
+                var singletonProps = options.TerminationMessage == null
+                    ? ClusterSingletonManager.Props(actorProps, clusterSingletonManagerSettings)
+                    : ClusterSingletonManager.Props(actorProps, options.TerminationMessage,
+                        clusterSingletonManagerSettings);
+
+                var singletonManagerRef = system.ActorOf(singletonProps, singletonName);
+
+                // create a proxy that can talk to the singleton we just created
+                // and add it to the ActorRegistry
+                if (createProxyToo)
+                {
+                    if (options.BufferSize != null)
+                    {
+                        singletonProxySettings = singletonProxySettings.WithBufferSize(options.BufferSize.Value);
+                    }
+
+                    CreateAndRegisterSingletonProxy<TKey>($"/user/{singletonManagerRef.Path.Name}", singletonManagerRef.Path.Name, singletonProxySettings, system, registry);
```

Comment:
Created a shared method for instantiating the proxies so we can use the same code in both methods.